src/marqo/tensor_search/utils.py

```diff
@@ -167,6 +168,10 @@ def parse_lexical_query(text: str) -> Tuple[List[str], List[str]]:
     All other terms go into blob, split by whitespace. blob starts as a string then splits into
     a list by space.
 
+    Backslash will be used to escape " or \, but if it is not followed by " or \,
```

Comment:
Is there any shared logic between how we parse a lexical search query and a filter string? If not, I think what you have is fine. If there is common logic, can we combine them?

Reply:
The issue with combining them is they have separate special characters, rules, DSLs, and parsers. The way we generate tokens for both is also slightly different.

Lexical query parsing is done here: https://github.com/marqo-ai/marqo/blob/18149c76bcb81089a0210158f2a6704aaa0f4498/src/marqo/tensor_search/utils.py#L163

Filter string parsing is done here: https://github.com/marqo-ai/marqo/blob/18149c76bcb81089a0210158f2a6704aaa0f4498/src/marqo/core/search/search_filter.py#L284

Reply:
Makes sense